sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -1001,11 +1082,11 @@
                         {
                             "type": "Microsoft.Authorization/roleAssignments",
                             "apiVersion": "2022-04-01",
-                            "name": "[guid(variables('materialization_identity_resource_id'), variables('offlineStoreConnectionTarget'), 'storage blob data contributor')]",
-                            "scope": "[variables('offlineStoreConnectionTarget')]",
+                            "name": "[guid(variables('materializationIdentity'), variables('offlineStore'), 'storage blob data contributor')]",
+                            "scope": "[variables('offlineStore')]",
                             "properties": {
                                 "roleDefinitionId": "[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')]",
-                                "principalId": "[if(empty(variables('materialization_identity_resource_id')), '', reference(variables('materialization_identity_resource_id'), '2023-01-31').principalId)]",
+                                "principalId": "[if(not(equals(parameters('materializationIdentityOption'), 'none')), reference(variables('materializationIdentity'), '2023-01-31').principalId, '')]",
```

Comment:
is it bretter to move this up in the conition of this deployment?

Reply:
Answer before, can't pass arm template validation check.